src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,113 @@
+package duke;
+import duke.commands.AddCommand;
+import duke.commands.ByeCommand;
+import duke.commands.Command;
+import duke.commands.Commands;
+import duke.commands.DateCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.FindCommand;
+import duke.commands.ListCommand;
+import duke.commands.TimeCommand;
+
+/**
+ * Parses user input to make sense of the command and find out what user wants.
+ */
+public class Parser {
+    private boolean entered = false;
+    private String description = "";
+    private Commands command;
+
+    /**
+     * Parses user input and finds out of user input is valid and if it is, which specific command is given.
+     *
+     * @param description User input containing command and tasks description.
+     * @return Specific command that user keyed in.
+     * @throws DukeException throw when error occurs.
+     */
+    public Command findCommand(String description) throws DukeException {
+        if (description.length() == 0 && entered) {
+            throw new DukeException("there's no commands");
+        } else if (description.equals("list")) {
+            command = Commands.LIST;
+            return new ListCommand();
+        } else if (description.length() >= 4 && description.substring(0, 4).equals("done")) {
```

Comment:
A line break could help to make this more readable. See "Place line break to improve readability" in the style guide. Some of the other `else if` statements in this method could also do with a line break.
On another note, this entire method is quite lengthy with many different cases, and it may be a good idea to try and break it up in future iterations.